### PR TITLE
Dapp browser API usage spec

### PR DESCRIPTION
**Pull request by vitvly**

### CODE REVIEW

File: docs/draft/webview.md
Comment (vkjr) [inline docs/draft/webview.md]: "It allows.. It injects.. it provides.."
It is a bit hard to keep track of what we are talking about. I'd suggest to replace few "it"s to "bridge" to make it more clear.

File: docs/draft/webview.md
Comment (vkjr) [inline docs/draft/webview.md]: For me as a newbie to API it would be really helpful to read some examples on when different API methods applicable.
For example it wasn't clear for me how `isStatus` used until I googled that metamask also injects `windows.ethereum` object and Dapp can differentiate ethereum provider this way.

File: docs/draft/webview.md
Comment (vkjr) [inline docs/draft/webview.md]: Could you please elaborate what type of connection this function means and why it is always true?

File: docs/draft/webview.md
Comment (vitvly) [reply docs/draft/webview.md]: Thanks @vkjr, not quite clear indeed. Fixed.

File: docs/draft/webview.md
Comment (vitvly) [reply docs/draft/webview.md]: Added notes for `isConnected`.

File: docs/draft/webview.md
Comment (vitvly) [reply docs/draft/webview.md]: Good point, will provide examples.

File: docs/draft/webview.md
Comment (andremedeiros) [inline docs/draft/webview.md]: `web3 support` -> `web3 compatible provider`

File: docs/draft/webview.md
Comment (andremedeiros) [inline docs/draft/webview.md]: Any chance this can be upstreamed?

Comment (andremedeiros) [reply docs/draft/webview.md]: Also do we have a method to keep up with upstream changes?

File: docs/draft/webview.md
Comment (vitvly) [reply docs/draft/webview.md]: Fixed.

File: docs/draft/webview.md
Comment (jakubgs) [inline docs/draft/webview.md:88]: Shouldn't the `(DEPRECATED)` be removed here?

**Comment by hesterbruikman** [comment]
What happens when a user withdraws permission? Is that something needs to be included here?

File: docs/draft/webview.md
Comment (vitvly) [reply docs/draft/webview.md]: Haven't yet updated our fork with upstream changes. I did rebase with latest upstream locally some time ago, but didn't tag it yet. Changes in our fork affect just a single file, so it is simple.

I've added a comment in their repo https://github.com/react-native-community/react-native-webview/pull/1099#issuecomment-611215560, but it's been quiet yet. Optimally - yes, we'd push a PR to their repo and close our fork.

File: docs/draft/webview.md
Comment (vitvly) [reply docs/draft/webview.md:88]: @jakubgs In latest EIP-119e release https://github.com/ethereum/EIPs/blob/master/EIPS/eip-1193.md `send` is marked as DEPRECATED to be replaced by `request`, if i understand correctly.

**Comment by vitvly** [comment]
@hesterbruikman need to investigate what happens then.

File: docs/draft/webview.md
Comment (jakubgs) [reply docs/draft/webview.md:88]: Fair enough.

File: docs/draft/webview.md
Comment (oskarth) [inline docs/draft/webview.md:8]: Missing permalink and number, see other specs for format

File: docs/draft/webview.md
Comment (oskarth) [inline docs/draft/webview.md]: Does it have to be a RN wrapper, or could it be generalized? On the spec level, now current app level

File: docs/draft/webview.md
Comment (oskarth) [inline docs/draft/webview.md]: perhaps be explicit about this being an app, and not just "Status" (official) app

File: docs/draft/webview.md
Comment (oskarth) [inline docs/draft/webview.md]: Would be good if we could lead with INTERFACE requirements and then talk about a specific implementation as an example, as opposed to the only way to implement the Status protocol.

File: docs/draft/webview.md
Comment (oskarth) [inline docs/draft/webview.md:83]: If these are deprecated, do we need to specify them? Considering this is the first time we are specifying it.

File: docs/draft/webview.md
Comment (oskarth) [inline docs/draft/webview.md:118]: What's the overlap between this and "EIP supported" spec?

File: docs/draft/webview.md
Comment (oskarth) [inline docs/draft/webview.md:127]: TODO

**Review by oskarth — CHANGES_REQUESTED** [review]
Thanks and good first spec PR!

One thing to keep in mind is to try to describe things at an interface/requirement level, and then talk about the current Status app as an implementation example.

Thought experiment:
- What would someone making a web tool / standalone Android app need to do to get equivalent API/protocol support as the current main Status app?

File: docs/draft/webview.md
Comment (oskarth) [inline docs/draft/webview.md:129]: Good reminder, some specs are missing this actually

File: docs/draft/webview.md
Comment (vitvly) [reply docs/draft/webview.md:83]: At least, they are still available in our implementation, probably because some Dapps might not work without them. Do you think this should be removed from the spec?

File: docs/draft/webview.md
Comment (vitvly) [reply docs/draft/webview.md]: Removed the mention of RN wrapper and modified wording elsewhere, so that it's more generic.

File: docs/draft/webview.md
Comment (vitvly) [reply docs/draft/webview.md:8]: Added permalink, but what should the number be?

File: docs/draft/webview.md
Comment (vitvly) [reply docs/draft/webview.md]: Fixed.

File: docs/draft/webview.md
Comment (vitvly) [reply docs/draft/webview.md]: Agree, made some changes, tried to be more generic.

File: docs/draft/webview.md
Comment (vitvly) [reply docs/draft/webview.md:118]: Deprecated methods are also listed in EIP-1193 spec. We do not implement `...Changed` events, for instance.

File: docs/draft/webview.md
Comment (decanus) [inline docs/draft/webview.md]: This isn't really an abstract, the abstract should explain what is contained in this specification.

**Review by decanus — CHANGES_REQUESTED** [review]

File: docs/draft/webview.md
Comment (vitvly) [reply docs/draft/webview.md]: Right! Added an abstract and created an Overview section.

File: docs/draft/webview.md
Comment (oskarth) [reply docs/draft/webview.md:8]: Depends on what order PRs are in :sweat_smile: I think 10 might be free? Check existing PRs, we can always change it at end before merge

File: docs/draft/webview.md
Comment (oskarth) [reply docs/draft/webview.md:83]: Hm ok, perhaps we can have a header called Not used` and then briefly mention them? I.e. we dont recommend X but some legacy apps might use

File: docs/stable/2-account.md
Comment (oskarth) [inline docs/stable/2-account.md]: What is this change?

**Review by oskarth — COMMENTED** [review]
Few outstanding comments but generally lgtm

File: docs/draft/webview.md
Comment (vitvly) [reply docs/draft/webview.md:8]: Modified permalink to include number.

File: docs/draft/webview.md
Comment (vitvly) [reply docs/draft/webview.md:83]: Added an `Unused` section and moved deprecated methods there.

File: docs/stable/2-account.md
Comment (vitvly) [reply docs/stable/2-account.md]: This is me rebasing against `develop` instead of `master`. Fixed.

**Review by oskarth — APPROVED** [review]

**Comment by flexsurfer** [comment]
we use MM library https://github.com/MetaMask/eth-phishing-detect/blob/master/src/config.json for detecting phishing, probably we should add this in the spec https://github.com/status-im/status-react/blob/a32c94b090bcbe96e3f52c48b1929a76d7c39e0a/src/status_im/browser/core.cljs#L74

**Comment by flexsurfer** [comment]
Tidus@discord
05/16/2020
on an unrelated note, @Embellished Tangible Darklingbeetle reported in #status that https://audius.co/ was blocked in the Web3 browser here in Status... I confirmed this, and verified that it doesn't occur on any other Web browsers on my iPhone (Brave, Chrome, Coinbase Wallet, DuckDuckGo, Edge, Firefox, Onion, Opera Touch, Safari, SnowHaze, Trust Wallet)... and am now wondering if Status actually somehow maintains a blacklist of domains or blocks them on the fly once they exhibit a certain type of behavior, and if that's in conflict with Principle II (censorship resistance) on https://status.im/about/#our-principles

https://snipboard.io/rGVETW.jpg

**Comment by oskarth** [comment]
Can we please get a second approval from Core so this can be merged? Discussed a week ago in Core Dev call.

@andremedeiros @vkjr @andremedeiros @flexsurfer @decanus  @status-im/status-core

**Review by vkjr — APPROVED** [review]

**Comment by oskarth** [comment]
Thanks @siphiuel and @vkjr!

**Comment by oskarth** [comment]
Ah right, seems like spell checker in CI (see log) doesn't allow this to be merged. New words have to be added to whitelist https://github.com/status-im/specs/blob/master/wordlist.txt. Could you have a log @siphiuel?

cc @decanus

**Comment by oskarth** [comment]
@siphiuel please make this pr mergeable, if you need help with spell checker you can ask @decanus

**Comment by vitvly** [comment]
@oskarth right! Fixed some spelling errors (`bidirection` -> `bidirectional`, and `acknowledgement` -> `acknowledgment`). For other words added a PR https://github.com/status-im/specs/pull/135.

**Comment by vitvly** [comment]
@oskarth @decanus fixed spell-check. @decanus - can you please review again?

File: docs/draft/webview.md
Comment (decanus) [inline docs/draft/webview.md]: Remove section if empty

**Review by decanus — CHANGES_REQUESTED** [review]

File: docs/draft/webview.md
Comment (vitvly) [reply docs/draft/webview.md]: @decanus removed empty sections

**Review by oskarth — APPROVED** [review]

File: docs/draft/webview.md
Comment (decanus) [inline docs/draft/webview.md]: "a proper environment" is kind of ambiguous

Comment (decanus) [inline docs/draft/webview.md]:
```suggestion
This document describes requirements that an application must fulfill in order to provide a proper environment for Dapps running inside a browser. A description of the Status Dapp API is provided, along with an overview of bidirectional communication underlying the API implementation. The document also includes a list of EIPs that this API implements.
```

**Review by decanus — CHANGES_REQUESTED** [review]

File: docs/draft/webview.md
Comment (vitvly) [reply docs/draft/webview.md]: Removed "facilities" as suggested.

**Comment by oskarth** [comment]
@decanus you have to review it, can't dismiss your review since you requested chanages

**Review by decanus — APPROVED** [review]

**Pull request merged**
